pkg/patch/patch_test.go

```diff
@@ -291,3 +298,228 @@ func TestGetOSVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestGeneratePatchedTag(t *testing.T) {
```

Comment:
I like this test! 

It is easy to understand from having worked with you this.

That test names could be even more explicit so as to differentiate between two types of tag that we are considering here in this test.

We have two tags

1. The user-supplied tag which leaves a breadcrumb indicating that the image was patched by copa 
2. The imagename tag which appears at the end of the image

If you use `ImageTag` and `UserSuppliedTag` in the test names a new contributor (or a curious end-user) would more easily understand this test.

Reply:
In the test case we are only supplying a user patched tag, which can either be empty `""`, or supplied `"example-supplied-tag"`. The differentiation is done within the function itself, not the parameters being passed in.

This is why for some of the test cases, an empty string is passed in but the expected result contains `defaultPatchedTagSuffix`